docs/client-sdk-states-and-lifecycle.mdx

```diff
@@ -0,0 +1,414 @@
+
+Once the client reaches this state, it stops tagging its telemetry pings with information about the experiment.
+
+
+### Telemetry
+
+Nimbus clients will submit telemetry via [Glean](https://mozilla.github.io/glean). The full set of experiment-related telemetry is defined in the [Nimbus Engineering Mana page](https://mana.mozilla.org/wiki/display/FJT/Nimbus+Engineering#NimbusEngineering-ExperimentTelemetry).
+
+Nimbus will use the [Glean Experiments API](https://mozilla.github.io/glean/book/user/experiments-api.html) to tag outgoing metrics pings with information about enrolled experiments:
+
+
+
+*   After initialization, the Nimbus SDK will list all known experiments in **Enrolled**, **Disqualified **or **WasEnrolled **state and call `setExperimentActive(slug, branch)` to tag outgoing telemetry pings with the enrolled branch of that experiment.
+*   When an experiment transitions to **Enrolled** state, the Nimbus SDK will call  `setExperimentActive(slug, branch)` to tag outgoing telemetry pings with the enrolled branch of the new experiment.
+*   When an experiment transitions from **WasEnrolled **to **Discarded**, the Nimbus SDK will call `setExperimentInactive(slug)` to cease tagging outgoing telemetry pings with information about the experiment.
+
+Nimbus will also emit Glean [events](https://mozilla.github.io/glean/book/user/metrics/event.html) on key experiment state transitions:
+
+
+
+*   An "**_enrollment_**" event, when an experiment enters the **Enrolled** state.
+    *   Event field "experiment" records the experiment slug.
+    *   Event field "branch" records the branch into which the client enrolled.
+    *   Event field "enrollmentId" contains a randomly-generated identifier for this enrollment.
+*   A "**_disqualification_**" event, when an experiment enters the **Disqualified **state.
+    *   Event field "experiment" records the experiment slug.
+    *   Event field "branch" records the branch into which the client enrolled.
+    *   Event field "enrollmentId" contains the enrollment id from the corresponding "enrollment" event.
+    *   Event field "reason" containing the reason for disqualification, as one of the following values:
+        *   "optout"
+        *   "targeting"
+*   An "**_unenrollment_**" event, when an experiment enters the **WasEnrolled **state.
+    *   Event field "experiment" records the experiment slug.
+    *   Event field "branch" records the branch into which the client enrolled.
+    *   Event field "enrollmentId" contains the enrollment id from the corresponding "enrollment" event.
+
+Nimbus will also emit Glean [events](https://mozilla.github.io/glean/book/user/metrics/event.html) when the client code calls experiment-related APIs:
+
+
+
+*   An "**_exposure_**" event, when client code calls `activateExperiment` or `isFeatureEnabled` for the first time, and the experiment is in **Enrolled** state.
```

Comment:
TODO: we don't actually have these in the current Nimbus SDK API.

question from the original doc:
> (@rfk) @travis79 do you know if these or equivalent are covered by the bugs you mentioned coming out of the API design discussion with James?

Reply:
It's possible to inform the SDK _not_ to send exposure events at that point, and to have them sent explicitly, right? I think that's worth explicating here.

Reply:
Now tracked in Jira SDK-299.